pkgs/by-name/bi/bitbox-bridge/package.nix

```
@@ -0,0 +1,46 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  libudev-zero,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "bitbox-bridge";
+  version = "1.6.1";
+
+  src = fetchFromGitHub {
+    owner = "BitBoxSwiss";
+    repo = "bitbox-bridge";
+    tag = "v${version}";
+    fetchSubmodules = true;
+    hash = "sha256-+pMXWXGHyyBx3N0kiro9NS0mPmSQzzBmp+pkoBLH7z0=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-6vD0XjGH1PXjiRjgnHWSZSixXOc2Yecui8U5FAGefBU=";
+
```

Should fix the failing darwin build

```suggestion
  useFetchCargoVendor = true;
  cargoHash = "sha256-6vD0XjGH1PXjiRjgnHWSZSixXOc2Yecui8U5FAGefBU=";

  postPatch = ''
    rm .cargo/config.toml
  '';
```

---

Applied the change, although have to mention its an odd change. I haven't checked the logs but I wonder why this happens.

---

Not odd actually, it is because of the configured linker for cargo

---

Could you elaborate?

---

`.cargo/config.toml` is configured with custom linkers for darwin builds. By removing it, we make cargo just use the default linker instead

---

ok but why? Shouldn't this removal be in the `rustPlatform.buildRustPackage` for darwin? If it should be kept, which dependency of this project causes a problem? This 4 lines seem redundant to me still, I'm just curious to understand it better ;)

---

Rarely there is some useful info in that config file and we are thinking about always removing it but we want to avoid causing more damage than good.